### PR TITLE
avoid including Poco headers globally

### PR DESCRIPTION
**Pull request by kejxu**
use forward declaration to avoid including Poco headers globally

### CODE REVIEW

**Comment by nuclearsandwich** [comment]
@kejxu what issue does this change resolve? I'm not opposed to it as written but knowing why it needs to be done this way will preserve the change into the future.

File: include/class_loader/class_loader_core.hpp [hunk 1]
```diff
@@ -41,12 +41,16 @@
...
-
 #include "class_loader/exceptions.hpp"
 #include "class_loader/meta_object.hpp"
 #include "class_loader/visibility_control.hpp"
 
+// forward declaration
+namespace Poco
+{
+  class SharedLibrary;
+}
```
Comment (wjwwood) [inline include/class_loader/class_loader_core.hpp]:
```suggestion
}  // namespace Poco
```

It's a nitpick, but I believe cpplint would complain here, though I don't know if we run class_loader through that linter off-hand.

**Review by wjwwood — COMMENTED** [review]
lgtm, but this should also take advantage of this change further (assuming no other header in this package still includes poco headers) to make poco-dev a build only depend, removing any build export depend on it, and ensuring the non-dev poco rosdep rule is the exec depend:

https://github.com/ros/rosdistro/blob/f614307d767ff062e1ec465c05105e72d1480c92/rosdep/base.yaml#L2652-L2667

**Comment by kejxu** [comment]
> @kejxu what issue does this change resolve? I'm not opposed to it as written but knowing why it needs to be done this way will preserve the change into the future.

Thanks for the question! the Poco headers used by ros.melodic includes `windows.h`, which could very likely lead to namespace corruption. It'd be safer to only the headers in .cpp files.

also thanks for improvement @wjwwood! (and adding the link to the styling guidelines, we've always wondered if there should be a styling guideline to follow)

**Pull request merged**